lib/model/autocomplete.dart

```diff
@@ -151,6 +151,20 @@ class AutocompleteViewManager {
   // void dispose() { … }
 }
 
+/// Provides the full list of possible items and filter predicate for
+/// `AutocompleteView` without having to deal with state management or
+/// result computation process.
+///
+/// Having this factored out of `AutocompleteView` helps in limiting
+/// it's responsibility to managing the autocomplete state and result
+/// computation process.
+abstract class AutocompleteDataProvider<T, Q extends AutocompleteQuery, R extends AutocompleteResult> {
```

Comment:
Reading @gnprice's review, I'm not sure if this was part of the factoring he suggested; Greg, what are your thoughts on it?

Reply:
Yeah, I think this can be simplified by making the two `AutocompleteDataProvider` methods be just abstract methods on `AutocompleteView` (like they are in the commit before this), and the concrete views implement those methods.

In the current revision, the split between AutocompleteDataProvider and AutocompleteView isn't really doing much to separate complexity because both of the AutocompleteView subclasses end up being pretty trivial, mostly just identifying how to construct the AutocompleteDataProvider. That is, virtually all the complexity lives on one side of the division, so the division isn't dividing much.